### ISSUE TITLE
How do you peoples get your angular website accepted by Google Adsense ?

### ISSUE THREAD

**Issue by borisBelloc** [scttcper/ng2-adsense#398]
Hello,

I don't know if I 'm allowed to ask here, but I don't know where to ask this question...

Every forums, thread, articles I read about "Google Adsense and Angular" says that Google's can't crawl angular website.

I tried to add `ng2-adsense` to my website but google told me that my website "got no content", google is not able to see what's inside my website.

I spent a lot of time trying to find a solution to make my Angular website Adsense compatible and didn't find any.

Can someone give me some tips ? 😷

**Comment by rawoke083** [comment]
I had to enable ssr or any pre-rendering framework.
From there I had to submit at least 3 times, and made sure to **show** them my "blog/articles" section. By then I had about 15 original pieces of content.

**Comment by scttcper** [comment]
I got mine approved without ssr, i did have to put more text and setup meta tags on all pages. make sure the pages you're adding ads on are public

**Comment by rawoke083** [comment]
Might be worthwhile to have a "Best Practises / FAQ" doc somewhere in this repo ? With regards to using Adsense+Angular+NG2-Adsense ?

* How to get approved (as you mentioned meta tags)
* Prerender vs SSR

**Comment by borisBelloc** [comment]
thanks you all for your answer, I didn't know about "Prerender" and those type of things.
Look like the solution is (Angular universal)[https://angular.io/guide/universal]

I know it's not related to ng2-adsense but like rawoke083 said a FAQ would help a lot of peoples

**Comment by scttcper** [comment]
You can see the angular website i got approved with here https://honbot.com/ I have others now, but only the first one has to be approved.

**Comment by mightymatth** [comment]
@scttcper I don't believe you 😆 I've tried to setup AdSense without SSR on site https://zorrotunes.com/, but I cannot get it approved for months. I've tried to prerender everything, setup dummy content, link it through, but it always says that there is not enough content. Do you have any suggestions?

**Comment by Ideki** [comment]
@mightymatth I had trouble with SSR too.
What I ended up doing was the following:

1. Prerendered my site
2. Copied the content of my prerended home page inside the <app-root></app-root> in my index.prod.html (don't forget your css too)
The dynamic content will replace the prerendered content after it has been loaded.
3. Published my side normally (no SSR, no prerender)
4. Submitted the application to AdSense again and got accepted.

I will eventually get around to get SRR/prerender working properly.

For now I just need to remember to repeat step 1 to 3 whenever I change my front page.

**Comment by kidqn** [comment]
> @mightymatth I had trouble with SSR too. What I ended up doing was the following:
> 
> 1. Prerendered my site
> 2. Copied the content of my prerended home page inside the  in my index.prod.html (don't forget your css too)
>    The dynamic content will replace the prerendered content after it has been loaded.
> 3. Published my side normally (no SSR, no prerender)
> 4. Submitted the application to AdSense again and got accepted.
> 
> I will eventually get around to get SRR/prerender working properly.
> 
> After you have been approved, you can remove the prerendered content (and css) from your index.prod.html

Hi @Ideki, you mean that remove all SSR, only generate prerender homepage , then submit to AdSense.
After approving, remove prerender and re-deploy SSR, right?
